contrib/IECoreUSD/src/IECoreUSD/VolumeAlgo.cpp

```diff
@@ -98,7 +99,7 @@ IECore::ObjectPtr readVolume( pxr::UsdVolVolume &volume, pxr::UsdTimeCode time,
 
 		SdfAssetPath fieldAssetPath;
 		fieldAsset.GetFilePathAttr().Get( &fieldAssetPath, time );
-		const std::string fieldFileName = fieldAssetPath.GetResolvedPath();
+		const std::string fieldFileName = DataAlgo::fromUSD( fieldAssetPath );
```

Comment:
Sorry, I should have spotted this before, but I think we could use `DataAlgo::fromUSD( fieldAsset.GetFilePathAttr() )` here, and then we wouldn't need the new addition to the DataAlgo API. This would also mean the "// Path resolution failed" handling in `dataFromSdfAssetPath()` would be in play.

Reply:
I think I got this right in https://github.com/ImageEngine/cortex/pull/1426/commits/88b4bfe95436a590a5064a184dba8c629ad27b90. And there are not any convenience methods to automatically return `StringDataPtr` instead of `DataPtr` + a cast, which I'm doing in `VolumeAlgo::readVolume()`?